### PR TITLE
DELETE Cache-Control: no-store clears locally cached blobs

### PR DESCRIPTION
**Pull request by josephmturner**
I'm not sure how best to test this behavior.

You can use `await blobs.get(id, { wait: false })` to check if a blob is locally stored ([docs](https://github.com/holepunchto/hyperblobs#const-content--await-blobsgetid-opts)) but I'm not sure if you want to expose that API in `hypercore-fetch`.

### CODE REVIEW

**Comment by josephmturner** [comment]
I added a test that I thought would work, but the commented portion just hangs.

File: index.js [hunk 1]
```diff
@@ -469,9 +469,41 @@ export default async function makeHyperFetch ({
...
     const pathname = decodeURI(ensureLeadingSlash(rawPathname))
+    const cacheControl = request.headers.get('Cache-Control') || ''
 
     const drive = await getDrive(`hyper://${hostname}/`, true)
 
+    if (cacheControl.includes('no-store')) {
+      // Delete locally cached blob, leaving entry metadata as-is.
+      if (pathname.endsWith('/')) {
+        // For directories, recursively clear cached blobs.
+        let didDelete = false
```
Comment (RangerMauve) [inline index.js:480]:
```suggestion
        let didClear = false
```

File: test.js [hunk 2]
```diff
@@ -276,6 +276,40 @@ test('DELETE a file', async (t) => {
...
+  const clearResponse = await fetch(file2URL, {
+    method: 'delete',
+    headers: {
+      'Cache-Control': 'no-store'
+    }
+  })
+
+  await checkResponse(clearResponse, t, 'Able to clear cache.')
+
+  const dirResponse = await fetch(created)
```
Comment (RangerMauve) [inline test.js:302]: Yeah testing this is kinda annoying. There's no direct API for it yet.

We'd need to use the hypercore.has api on the blobs hypercore.

https://github.com/holepunchto/hypercore#const-has--await-corehasstart-end

I think we can add a TODO for that and trust that .clear worked for now

**Review by RangerMauve — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (josephmturner) [reply index.js:480]: Fixed. Thanks!

File: test.js (hunk 2 above)
Comment (josephmturner) [reply test.js:302]: Commented out the last request.

**Comment by josephmturner** [comment]
Closes #66.

**Comment by josephmturner** [comment]
Merged into https://git.sr.ht/~ushin/hypercore-fetch-ushin/commit/815b1d1b82298dcc6b0f77c26a8eb721dc9b7456

Thank you for your the code review!

**Pull request closed without merging**